src/events/guildDelete.js

```diff
@@ -0,0 +1,15 @@
+const Event = require("../structures/Event");
+
+class GuildDelete extends Event {
+    constructor(...args) {
+        super(...args);
+    }
+
+    async execute(guild) {
+        if (!guild.available) return;
+
+        this.client.handlers.tasks.create('deleteSettings', Date.now() + 604800000, { guild: guild.id });
```

Comment:
why does the task need to wait for 7 days? i know it's the max discord requires, but why not do just 24 hours? if they are not adding it back by then, why do we bother to store this data for that long?

Reply:
There's the potential that 24 hours isn't enough. While I agree that most users will likely re-add the bot within the first 24 hours if there was a mistake, storing the data doesn't cost us that much in terms of resources. In short, convenience for the user is why I think it can wait.